js/embed.js

```diff
@@ -7,50 +7,68 @@ import { ErrorView } from './ErrorView'
 import 'font-awesome/css/font-awesome.css'
 import './widgets.css'
 
-export async function renderWidgets(baseUrl, wsUrl, loader) {
-    let connectionInfo = ServerConnection.makeSettings({
-        baseUrl,
-        wsUrl
-    });
 
-    const kernelSpecs = await Kernel.getSpecs(connectionInfo)
+export class WidgetApplication {
+    constructor (baseUrl, wsUrl, loader) {
+        this._baseUrl = baseUrl
+        this._wsUrl = wsUrl
+        this._loader = loader
+    }
 
-    console.log(`Starting kernel ${kernelSpecs.default}`)
+    async renderWidgets() {
+        var baseUrl = this._baseUrl
```

Comment:
had some compilation issues using this._baseUrl and this._wsUrl in the makeSettings call, so I used local variables instead (if there's a better js way to do this that, I can clean these lines up!)

Reply:
When you write `{ baseUrl, wsUrl }`, this is just syntactic sugar for:

```javascript
{ 
  'baseUrl': baseUrl, 
  'wsUrl': wsUrl 
}
```

ie defining a JavaScript object with keys `baseUrl` and `wsUrl`, and values bound to the value of `baseUrl` and `wsUrl` in scope.

Thus, if you write:

```javascript
{ this._baseUrl, this._wsUrl }
```

that desugars to:

```javascript
{ 
  'this._baseUrl': this._baseUrl, 
  'this._wsUrl': this._wsUrl 
}
```

... basically the keys don't make sense (and babel complains that the keys contain the reversed keyword `this`). I made it work with:

```javascript
        const connectionInfo = ServerConnection.makeSettings({
            baseUrl: this._baseUrl,
            wsUrl: this._wsUrl
        });
```

`ServerConnection.makeSettings` expects an object with keys `baseUrl` and `wsUrl`.

Note that that also changes the `let` to a `const`, since the reference doesn't change.

Generally, when I want to understand how ES6 features translate to 'normal' JavaScript, I find [this console](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=MYewdgzgLgBA7hAqgJwDYwLwwEQEMBGw2AUKJLPrhAKYrpbYAm1AZiaeBCKtQHSogA5gAoA3jEo06AGnhI0MAL4BKIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=false&presets=es2015%2Cstage-0%2Cstage-2%2Cstage-3&prettier=false&targets=&version=6.26.0&envVersion=) useful.

Hope this makes some sense.

Reply:
yes, that makes a lot of sense! Thank you for taking the time to break it down, and for the console link! 👍  I'll clean up that section so that I'm not making local variables and update the PR!